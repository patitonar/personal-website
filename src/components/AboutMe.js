import React from 'react';
import { StyleSheet, css } from 'aphrodite';
import Paper from 'material-ui/Paper';
import image from '../assets/image.jpg';

const imageUrl = `url('${image}')`;

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#009688',
    color: 'white',
    '@media (max-width: 1024px)': {
      paddingTop: 20,
      paddingBottom: 20,
    },
  },
  content: {
    fontSize: 'large',
    display: 'flex',
    '@media (max-width: 1024px)': {
      alignItems: 'center',
      flexDirection: 'column',
      width: '80%',
    },
    '@media (min-width: 1024px)': {
      width: '60%',
    },
  },
  h1: {
    fontWeight: 'normal'
  },
  avatar: {
    '@media (max-width: 599px)': {
      height: 100,
      width: 100,
      margin: 10,
    },
    '@media (min-width: 600px)': {
      height: 150,
      width: 150,
      margin: '20px 40px 20px 20px',
    },
    flex: '1 0 auto',
    textAlign: 'center',
    backgroundImage: imageUrl,
    backgroundPosition: 'center',
  },
  avatarWrapper: {
    height: 'inherit',
    display: 'flex',
    alignItems: 'center'
  },
});


const AboutMe = () => (
  <div className={css(styles.container)}>
    <h1 className={css(styles.h1)}>About Me</h1>
    <div className={css(styles.content)}>
      <div className={css(styles.avatarWrapper)}>
        <Paper className={css(styles.avatar)} zDepth={3} circle={true} />
      </div>
      <div>
        <p>
          Hi, I'm Gerardo. I'm a Software Engineer from Argentina graduated at UTN FRSF, also a frustrated gamer and a football fan.
          I love researching cutting edge technologies and playing around with them.
        </p>
        <p>
          I have experience in web development. I'm currently learning and gaining experience with React and its ecosystem.
          I develop great web applications writing elegant code, making focus on good UI/UX, performance, maintainability
          and working with tools that help me being productive.
        </p>
      </div>
    </div>
  </div>
);

export default AboutMe;
