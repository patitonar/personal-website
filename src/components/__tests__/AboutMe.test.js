import React from 'react';
import renderer from 'react-test-renderer';
import AboutMe from '../AboutMe';

it('AboutMe renders correctly', () => {
  const tree = renderer.create(
    <AboutMe />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
