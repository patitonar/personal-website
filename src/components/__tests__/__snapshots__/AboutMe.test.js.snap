// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe renders correctly 1`] = `
<div
  className="css-16zp9vm"
>
  <h1
    className="css-15q2ai2"
  >
    About Me
  </h1>
  <div
    className="css-iahclo"
  >
    <div
      className="css-5n2qub"
    >
      <div
        className="css-1xiydre"
      />
    </div>
    <div>
      <p>
        I'm a Software Engineer from Argentina, also a frustrated gamer and a football fan. I love researching cutting edge technologies and playing around with them.
      </p>
      <p>
        I have experience working as a full stack developer, although frontend development is what I really like. Some of the technologies I worked with are PHP, Java and nodejs in the backend. On the frontend I built some apps using backbone, angular and react. Also, Ionic for mobile development and React-VR just for VR fun.
      </p>
      <p>
        I'm really good at learning new techs and I always try to build great applications writing elegant code, making focus on good UI/UX, performance, maintainability and working with tools that help me be productive. Also I like contributing to open source projects whenever I have the opportunity.
      </p>
    </div>
  </div>
</div>
`;
